operators/src/processing/vector_join/equi_data_join.rs

```diff
@@ -0,0 +1,401 @@
+        left: &FeatureCollection<G>,
+        right: &DataCollection,
+    ) -> Result<FeatureCollection<G>>
+    where
+        D1: DataRef<'i, T> + 'i,
+        D2: DataRef<'i, T> + 'i,
+        T: 'i + PartialEq,
+    {
+        let matches_iter = left_join_values
+            .as_ref()
+            .iter()
+            .zip(left.time_intervals())
+            .map(move |(left_value, left_time_interval)| {
+                right_join_values
+                    .as_ref()
+                    .iter()
+                    .zip(right.time_intervals())
+                    .map(|(right_value, right_time_interval)| {
+                        (left_value == right_value)
+                            && left_time_interval.intersects(right_time_interval)
+                    })
+                    .collect::<Vec<bool>>()
+            });
+
+        let mut builder = FeatureCollection::<G>::builder();
+
+        // create header by combining values from both collections
+        for (column_name, column_type) in left
+            .column_types()
+            .into_iter()
+            .chain(right.column_types().into_iter())
+        {
+            builder.add_column(column_name, column_type)?;
+        }
+
+        let mut builder = builder.finish_header();
+
+        for ((left_feature_idx, matches), geometry) in matches_iter
+            .enumerate()
+            .zip(left.geometries().map(Into::into))
+        {
+            // TODO: is it better to traverse column-wise?
+
+            for (right_feature_idx, matched) in matches.into_iter().enumerate() {
+                if !matched {
+                    continue;
+                }
+
+                // add geo
+                builder.push_geometry(geometry.clone())?;
+
+                // add time
+                builder.push_time_interval(
+                    left.time_intervals()[left_feature_idx]
+                        .intersect(&right.time_intervals()[right_feature_idx])
+                        .expect("must intersect for join"),
+                )?;
+
+                // add left values
+                for column_name in left.column_names() {
+                    let feature_data = left
+                        .data(column_name)
+                        .expect("must exist")
+                        .get_unchecked(left_feature_idx);
+                    builder.push_data(column_name, feature_data)?;
+                }
+
+                // add right values
+                for column_name in right.column_names() {
+                    let feature_data = right
+                        .data(column_name)
+                        .expect("must exist")
+                        .get_unchecked(right_feature_idx);
+                    builder.push_data(column_name, feature_data)?;
+                }
+
+                builder.finish_row();
+            }
+        }
+
+        builder.build().map_err(Into::into)
+    }
+
+    /// Fix column by renaming column with duplicate keys
```

Comment:
maybe it would be wiser to directly use the correct column names (fixing them earlier) instead of creating a new collection here.

Reply:
How would I do this?

Reply:
7d53833